### PR TITLE
make --check-conflicts report source of conflicts

### PR DESCRIPTION
**Pull request by boegel**
fix for #1816

@verdurin: how does this look?

```
$ eb --check-conflicts sympy-0.7.2-goolf-1.4.10-Python-2.7.3.eb Seaborn-0.6.0-goolf-1.4.10-Python-2.7.5.eb Python-2.7.6-goolf-1.4.10.eb
== temporary log file in case of crash /var/folders/xl/p7lmsxcs0bsg5_zv9y9lzk5c0000gn/T/eb-9gdqhJ/easybuild-3aUchF.log
Conflict between (dependencies of) easyconfigs: Python-2.7.3-goolf-1.4.10 vs Python-2.7.5-goolf-1.4.10 
	Python-2.7.3-goolf-1.4.10 as dep of: sympy-0.7.2-goolf-1.4.10-Python-2.7.3
	Python-2.7.5-goolf-1.4.10 as dep of: Seaborn-0.6.0-goolf-1.4.10-Python-2.7.5
Conflict between (dependencies of) easyconfigs: Python-2.7.3-goolf-1.4.10 vs Python-2.7.6-goolf-1.4.10 
	Python-2.7.3-goolf-1.4.10 as dep of: sympy-0.7.2-goolf-1.4.10-Python-2.7.3
Conflict between (dependencies of) easyconfigs: Python-2.7.5-goolf-1.4.10 vs Python-2.7.6-goolf-1.4.10 
	Python-2.7.5-goolf-1.4.10 as dep of: Seaborn-0.6.0-goolf-1.4.10-Python-2.7.5
Conflict between (dependencies of) easyconfigs: zlib-1.2.7-goolf-1.4.10 vs zlib-1.2.8-goolf-1.4.10 
	zlib-1.2.7-goolf-1.4.10 as dep of: Python-2.7.3-goolf-1.4.10, Python-2.7.6-goolf-1.4.10, sympy-0.7.2-goolf-1.4.10-Python-2.7.3
	zlib-1.2.8-goolf-1.4.10 as dep of: Python-2.7.5-goolf-1.4.10, SQLite-3.8.5-goolf-1.4.10, Seaborn-0.6.0-goolf-1.4.10-Python-2.7.5, Tcl-8.6.4-goolf-1.4.10
ERROR: One or more conflicts detected!
```

### CODE REVIEW

**Comment by boegel** [comment]
@wpoely86 please review?

File: easybuild/tools/robot.py [hunk 1]
```diff
@@ -81,46 +81,66 @@ def mk_key(spec):
...
 
-    def is_conflict((name, installver), (name1, installver1), (name2, installver2)):
+            deps_for[key] = (sorted(nub(deps_for[key][0])), sorted(nub(deps_for[key][1])))
+
+            # also track reverse deps (except for ghost entry)
+            if key != (None, None):
+                for dep in build_deps + runtime_deps:
+                    dep_of.setdefault(dep, set()).add(key)
+
+    def check_conflict((name, installver), dep1, dep2):
```
Comment (wpoely86) [inline easybuild/tools/robot.py]: doc the arguments

**Comment by verdurin** [comment]
This looks good to me.

**Comment by boegel** [comment]
Thanks for the feedback @verdurin, and for the review @wpoely86!

Going in...

**Pull request merged**